Automattic/wp-calypso
# A4A-Partner Directory: Update the Agency Details page

Resolves: https://github.com/Automattic/automattic-for-agencies-dev/issues/663

## Proposed Changes

![image](https://github.com/Automattic/wp-calypso/assets/9832440/1f769ec9-a276-44dc-b949-def7107a82cb)

![image](https://github.com/Automattic/wp-calypso/assets/9832440/a1ef6b5c-d1c3-429a-bdab-aeceb373b79c)

This PR fixes these issues:

- The Agency Details header, in the description `Want to update your expertise instead?` is a link that will take you to the Expertise form.
- The Company location is a country selector instead of a textbox.
- The Minimum project budget list is a selector instead of a textbox.
- The Budget details Optionally set your minimum budget. Clients can filter these details to find the right agency should be under the title.
- The Availability checkbox should be a switcher.
- The CTA buttons, like the `Publish public profile` should be aligned on the right.
- The dropdown selector should be larger.
- The `Publish public profile` is only active if you fill out all the fields and select at least one item on each selector.

The form selector hook will be used to map the API data and the form. There, the selector values will be mapped by their slugs.

This PR is not connected with the endpoint, so the form won't do anything. It will be addressed in a follow-up PR together with this: https://github.com/Automattic/wp-calypso/pull/91683

Any style/design issue will be addressed after the designer's review (behind feature/section) flag.

There are known issues to be fixed. The services, products, and languages selectors are working with value instead of slug. 

## Why are these changes being made?


*

## Testing Instructions

- Check the code
- Comparing with trunk:
   - Now, the Minimum project budget is a selector instead of a textbox.
   - The `Optionally set your minimum budget. Clients can filter these details to find the right agency.` is below the `Budget details` title.
   - The Availability is a switcher instead of a checkbox. Bonus: If you change the status, the text will change: `I'm accepting new clients` <=> `I'm not accepting new clients`.
   - The dropdown selectors are a bit larger (height).

## Pre-merge Checklist
